server/src/main/java/umm3601/Server.java

```diff
@@ -10,6 +10,12 @@
 public class Server {
     public static void main(String[] args) throws IOException {
 
+        // This users looks in the folder `public` for the static web artifacts,
+        // which includes all the HTML, CSS, and JS files generated by the Angular
+        // build. This `public` directory _must_ be somwhere in the classpath;
+        // a problem which is resolved in `server/build.gradle`.
+        staticFiles.location("/public");
```

Comment:
Actually, I totally forgot about the routing stuff and, as you say, didn't change a thing.

I just checked, and everything seems to work fine in dev mode, including things like live reloading when I make changes in IDEA. I really don't understand how this is happening, though. When I run from the start script in the deployed code and go to `<hostname>:4567` that returns `index.html` but leaves me on port 4567. When I do that in dev mode, it switches me to port 9000.

I haven't clue one about what part of the world is being smart enough to "manage" that. Do either @joethe or @devshawn have any idea what's making that work the "right way" for us now?